### PR TITLE
feat: add source files API functionality

### PR DESCRIPTION
**Pull request by vorobeyme**

### CODE REVIEW

File: crowdin/source_files.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+// Before adding source files to the project, upload each file to the Storage first.
+//
+// Note: If you use branches, make sure your master branch is the first one you integrate with Crowdin.
+//
+// Crowdin API docs: https://developer.crowdin.com/api/v2/#tag/Source-Files
+type SourceFilesService struct {
+	client *Client
+}
+
+// ListBranches returns a list of project branches.
```
Comment (andrii-bodnar) [inline crowdin/source_files.go:22]: Also include the String-based related branch methods in the `source_files` section.

https://developer.crowdin.com/api/v2/string-based/#tag/Branches

File: crowdin/source_files.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+	return res.Data, resp, err
+}
+
+// UpdateRestoreFile updates a file in the project or restores it to one
+// of the previous revisions.
+// For updating the file, use the `storageId` body parameter.
+// For restoring the file, use the `revisionId` body parameter.
+//
+// https://developer.crowdin.com/api/v2/#operation/api.projects.files.put
+func (s *SourceFilesService) UpdateRestoreFile(ctx context.Context, projectID, fileID int64, req *model.FileUpdateRestoreRequest) (
```
Comment (andrii-bodnar) [inline crowdin/source_files.go:215]:
```suggestion
func (s *SourceFilesService) UpdateOrRestoreFile(ctx context.Context, projectID, fileID int64, req *model.FileUpdateRestoreRequest) (
```

File: crowdin/source_files.go (hunk 1 above)
Comment (vorobeyme) [reply crowdin/source_files.go:22]: Is it possible to move all brunch-related methods to a separate service (e.g. Branches)?

Comment (andrii-bodnar) [reply crowdin/source_files.go:22]: Let's move it to a separate service

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**